nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -259,6 +261,8 @@ public Object run() {
                             ? putFlowFile.getAttribute(CoreAttributes.FILENAME.key()) + codec.getDefaultExtension()
                             : putFlowFile.getAttribute(CoreAttributes.FILENAME.key());
 
+                    final FsPermission permission = FsCreateModes.applyUMask(FsPermission.getFileDefault(), FsPermission.getUMask(configuration));
```

Comment:
I suspect that calculating permissions this way could problematic; I'm unaware if all FileSystem implementations use FsPermission.getFileDefault() as the value against which the umask is applied.

While reviewing an earlier version of your PR that did not contain changes related to this line of additional code, I was able to test various use-cases against a real HDFS cluster and get the correct results from newly created files, but unable to get the file permission test to pass when using various umask values (or lack of values) in the processor configuration and hdfs-site.xml.  I think this is more a symptom of the MockFileSystem implementation, rather than an issue with PutHDFS itself, outside of including the `fs.permissions.umask-mode`/default umask.

The only thing that should be changed in PutHDFS should be the application of the umask from the processor properties, or using the Hadoop config's `fs.permissions.umask-mode` with a fall-back to the default umask, by using `FsPermission.getUmask(config)`.

I'm hesitant to merge this kind of change, since I was able to get proper results from a live HDFS cluster without it, and no JIRAs against PutHDFS have been filed since the resolution of https://issues.apache.org/jira/browse/NIFI-4311, other than using the `fs.permissions.umask-mode` property.